Ghidra/Features/PDB/src/pdb/cpp/iterate.cpp

```diff
@@ -280,137 +269,135 @@ void dumpFunctionStackVariables( DWORD rva )
 
-		dumpFunctionStackVariables(address);
-		dumpFunctionLines(pSymbol, pSession);
+		dumpFunctionStackVariables(ctx, address);
+		dumpFunctionLines(*pSymbol, ctx.Session());
 
-		printf("%s</function>\n", indent(8));
+		printf("%S</function>\n", indent(8).c_str());
 
 		pSymbol = 0;
 	}
-	printf("%s</functions>\n", indent(4));
+	printf("%S</functions>\n", indent(4).c_str());
 }
 
 void iterateSymbolTable(IDiaEnumSymbols * pSymbols) {
-	DWORD celt;
-	IDiaSymbol * pSymbol;
-
+	DWORD celt = 0;
 	while ( 1 ) {
-		if (pSymbols->Next( 1, &pSymbol, &celt ) < 0) {
+		CComPtr<IDiaSymbol> pSymbol;
+		if (FAILED(pSymbols->Next( 1, &pSymbol, &celt ))) {
 			break;
 		}
 		if (celt != 1) {
 			break;
 		}
 
-		BSTR name = getName(pSymbol);
-		if (name == NULL) {
+		const std::wstring name = getName(*pSymbol);
+		if (name.empty()) {
 			continue;
 		}
-
-		printf("%s",                    indent(12));
-		printf("<symbol name=\"%ws\" ", name);
-		printf("address=\"0x%x\" ",     getRVA(pSymbol));
-		printf("length=\"0x%I64x\" ",   getLength(pSymbol));
-		printf("tag=\"%ws\" ",          getTagAsString(pSymbol));
-		printf("kind=\"%ws\" ",         getKindAsString(pSymbol));
-		printf("index=\"0x%x\" ",       getIndex(pSymbol));
-		printf("undecorated=\"%ws\" ",  getUndecoratedName(pSymbol));
-		printf("value=\"%ws\" ",        getValue(pSymbol));
-		printf("datatype=\"%ws\" ",     getTypeAsString(pSymbol));
+		printf("%S",                    indent(12).c_str());
+		printf("<symbol name=\"%S\" ",  name.c_str());
+		printf("address=\"0x%x\" ",     getRVA(*pSymbol));
+		printf("length=\"0x%I64x\" ",   getLength(*pSymbol));
+		printf("tag=\"%S\" ",           getTagAsString(*pSymbol).c_str());
+		printf("kind=\"%S\" ",          getKindAsString(*pSymbol).c_str());
+		printf("index=\"0x%x\" ",       getIndex(*pSymbol));
+		printf("undecorated=\"%ws\" ",  getUndecoratedName(*pSymbol).c_str());
+		printf("value=\"%S\" ",         getValue(*pSymbol).c_str());
+		printf("datatype=\"%S\" ",      getTypeAsString(*pSymbol).c_str());
 		printf(" />\n");
-
-		pSymbol = NULL;
 	}
 }
 
 void iterateSourceFiles(IDiaEnumSourceFiles * pSourceFiles) {
-	HRESULT hr;
-	DWORD celt;
-	IDiaSourceFile * pSourceFile;
+	HRESULT hr = S_OK;
+	DWORD celt = 0;
+    CComPtr<IDiaSourceFile> pSourceFile;
 	while ( SUCCEEDED( hr = pSourceFiles->Next( 1, &pSourceFile, &celt ) ) && celt == 1 ) {
-		BSTR name;
-		pSourceFile->get_fileName( &name );
-		DWORD id;
+		BSTR temp = NULL;
+		pSourceFile->get_fileName( &temp );
+		bstr_t name;
+		name.Attach(temp);
+		DWORD id = 0;
 		pSourceFile->get_uniqueId( &id );
-		if ( name != NULL ) {
-			printf("%s<source_file name=\"%ws\" id=\"0x%x\" /> \n", indent(12), name, id);
+		if ( name.GetAddress() != NULL ) {
+			printf("%S<source_file name=\"%ws\" id=\"0x%x\" /> \n", indent(12).c_str(), name.GetBSTR(), id);
```

Comment:
Note difference from comment made for line 319 regarding %ws and GetBSTR().  Which is correct?